mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/TaskExecutorState.java

```diff
@@ -251,8 +252,11 @@ TaskExecutorRegistration getRegistration() {
         return this.registration;
     }
 
-    protected CompletableFuture<TaskExecutorGateway> getGateway() {
-        return this.gateway;
+    protected TaskExecutorGateway getGateway() throws ExecutionException, InterruptedException {
+        if (this.gateway == null) {
+            throw new IllegalStateException("gateway is null");
```

Comment:
isn't NullPointerException semantically more correct?

Reply:
NPE tends to be harder to read in logs as it doesn't point to the exact null object. Semantically this is illegal state too if a "client" try to get the gateway while it's has failed/disconnected?